pyfr/plugins/base.py

```diff
@@ -167,3 +169,35 @@ def __init__(self, intg, *args, **kwargs):
 
             if not isinstance(eidxs, slice):
                 self._ele_region_data[f'{etype}_idxs'] = eidxs
+
+
+class SurfaceMixin:
+    def __init__(self, intg, *args, **kwargs):
+        super().__init__(intg, *args, **kwargs)
+
+        self._surf_quad = {}
+
+    def surf_quad(self, itype, proj, *args):
```

Comment:
Just make this a quasi-static method and avoid the need for a new member variable `_surf_quad`.

Reply:
Or you can @memoize this (but remove *args).

Reply:
I added the `*args` so you could just pass `eles.basis.faces[fidx]` but I think this is a niche use case.

Reply:
Yeah, it is usually just asking for trouble.